LuaStack::read to std::string fails

I've been using this library for a few days and i'm enjoying it so far. However, i've found one problem that appears when compiling with GCC 4.7.2 (haven't been able to test on other compilers, sorry):

``` c++
std::string hello = state["mystr"]
```

It outputs this error when compiling:

```
In function `std::string lua::Value::to<std::string>() const':
LuaValue.h:185: undefined reference to `std::string lua::stack::read<std::string>(lua_State*, int)'
```

So i've just added this to LuaStack.h, right after the `const char*` specialization:

``` c++
template<>
inline std::string read(lua_State* luaState, int index) {
        return lua_tostring(luaState, index);
}
```

And now it works without any apparent problem. I suspect this bug is caused by my compiler, which is not able to transform the `const char*` specialization to an `std::string`.

I have not used any function from the library except for simple value retrievals, so i don't know if this bug has furthest reach; just in case i open this issue to address those problems if they occur.

Thanks for your work, i really enjoy working with this library!

--- a/include/LuaStack.h
+++ b/include/LuaStack.h
@@ -383,6 +383,11 @@ namespace lua { namespace stack {
     inline const char* read(lua_State* luaState, int index) {
         return lua_tostring(luaState, index);
     }
+    
+    template<>
+    inline std::string read(lua_State* luaState, int index) {
+        return lua_tostring(luaState, index);
+    }
 
     template<>
     inline const unsigned char* read(lua_State* luaState, int index) {

